subsystems/shootersubsystem.py

```diff
@@ -0,0 +1,170 @@
+from commands2 import Subsystem
+from wpilib import SmartDashboard
+from wpimath.geometry import Rotation2d
+from phoenix6.configs import CurrentLimitsConfigs
+from util.simtalon import Talon
+from util.simneo import NEOBrushless
+from util.simcoder import CTREEncoder
+import constants
+
+
+class ShooterSubsystem(Subsystem):
+    def __init__(self) -> None:
+        Subsystem.__init__(self)
+        self.setName(__class__.__name__)
+
+        self.angleMotor = Talon(
+            constants.kAngleMotorCANId,
+            constants.kAngleMotorName,
+            constants.kAngleMotorPGain,
+            constants.kAngleMotorIGain,
+            constants.kAngleMotorDGain,
+            constants.kAngleMotorInverted,
+        )
+        self.leftShootingMotor = NEOBrushless(
+            constants.kLeftShootingMotorCANId,
+            constants.kLeftShootingMotorName,
+            constants.kLeftShootingMotorPIDSlot,
+            constants.kLeftShootingMotorPGain,
+            constants.kLeftShootingMotorIGain,
+            constants.kLeftShootingMotorDGain,
+            constants.kLeftShootingMotorInverted,
+            constants.kLeftShootingMotorKv,
+        )
+        self.rightShootingMotor = NEOBrushless(
+            constants.kRightShootingMotorCANId,
+            constants.kRightShootingMotorName,
+            constants.kRightShootingMotorPIDSlot,
+            constants.kRightShootingMotorPGain,
+            constants.kRightShootingMotorIGain,
+            constants.kRightShootingMotorDGain,
+            constants.kRightShootingMotorInverted,
+            constants.kRightShootingMotorKv,
+        )
+
+        self.shooterEncoder = CTREEncoder(constants.kShooterAngleEncoderCANId, 0)
+
+        self.leftShootingMotor.setSmartCurrentLimit(
+            constants.kShootingMotorCurrentLimit
+        )
+        self.rightShootingMotor.setSmartCurrentLimit(
+            constants.kShootingMotorCurrentLimit
+        )
+
+        self.angleMotor.setCurrentLimit(
+            CurrentLimitsConfigs().with_stator_current_limit(
+                constants.kAngleMotorCurrentLimit
+            )
+        )
+
+        # set motor position, encoder is inverse of the shooter
+
+        self.angleMotor.setEncoderPosition(
+            self.shooterEncoder.getPosition().radians()
+            / constants.kRadiansPerRevolution
+            * constants.kAngleMotorRatio
+            * -1
+        )
+
+        self.targetAngle = Rotation2d()
+        self.leftTargetSpeed = 0
+        self.rightTargetSpeed = 0
+
+        # speeds are in RPM, same as velocity control mode
+
+    def setShooterAngle(self, angle: Rotation2d) -> None:
+        self.targetAngle = min(
+            max(
+                constants.kShooterMinAngle,
```

Comment:
what should be taken into account here is that we may want to fudge beyond the range of these angles as a result of bad startup / weird encoder nonsense
I'd add the fudge with this